ue5osc/__init__.py

```diff
@@ -1,21 +1,123 @@
+        return values
 
+    def get_player_location(self, print_val: bool = False) -> list[float, float, float]:
+        """Returns x, y, z location of the player in the Unreal Environment and optionally prints."""
+        values = self.send_and_wait("/get/location")
+        if print_val:
+            self.print_values("location", values)
+        return values
 
-class OSCSender:
-    def __init__(self, ip: str, port: int):
-        self.client = udp_client.SimpleUDPClient(ip, port)
+    def set_player_location(self, x: float, y: float, z: float):
+        """Sets X, Y, and Z values of an Unreal Camera."""
+        self.client.send_message("/set/location", [x, y, z])
+
+    def get_player_rotation(self, print_val: bool = False) -> list[float, float, float]:
+        """Returns pitch, yaw, and roll and can optionally print."""
+        values = self.send_and_wait("/get/rotation")
+        if print_val:
+            self.print_values("rotation", values)
+        return values
+
+    def set_player_yaw(self, yaw: float, pawn: int = 0):
+        """Set the camera yaw in degrees."""
+        ue_roll, ue_pitch, _ = self.get_player_rotation()
+        self.client.send_message("/set/rotation", [ue_pitch, ue_roll, yaw])
+
+    def __rotate(self, rotation: float, pawn: int = 0):
+        """Rotate player a number of degrees."""
+        ue_roll, ue_pitch, ue_yaw = self.get_player_rotation()
+        yaw = float(ue_yaw) + rotation
+        self.client.send_message("/turn/left", yaw)
 
     def move_forward(self, amount: float):
+        """Move robot forward."""
         self.client.send_message("/move/forward", float(amount))
 
     def turn_left(self, degree: float):
+        """Turn robot left."""
         self.client.send_message("/turn/left", float(degree))
 
     def turn_right(self, degree: float):
+        """Turn robot right."""
         self.client.send_message("/turn/right", float(degree))
 
+    def move_backward(self, amount: float):
+        """Moverobot backwards."""
+        self.client.send_message("/move/forward", float(-amount))
+
+    def save_image(self) -> None:
+        """Takes screenshot with the default name"""
+        file_path = self.path / f"{self.img_number:06}"
+        self.img_number += 1
+        self.client.send_message("/save/image", file_path)
+        sleep(1)
+
+    def request_image(self) -> bytes:
+        """Requests the image we saved."""
+        image = Image.open(self.file_path)
+        return image
+
+    def show(self):
+        """If matplotlib is being used, show the image taken to the plot"""
+        import matplotlib.pyplot as plt
+
+        plt.imshow(self.request_image())
+
     def take_screenshot(self, filename: str):
+        """Save a screenshot with a unique name"""
         self.client.send_message("/screenshot", filename)
 
     def reset_to_start(self):
+        """Reset agent to the start location using a UE Blueprint command."""
         self.client.send_message("/reset", 0.0)

```

Comment:
Should this be a `send_and_wait`? Or is the reset fast enough to be considered instantaneous?

Reply:
It normally is instantaneous; however, I added a sleep right after it just to ensure.